src/Mvc/Mvc.Core/src/Infrastructure/VirtualFileResultExecutor.cs

```diff
@@ -56,11 +60,27 @@ public virtual Task ExecuteAsync(ActionContext context, VirtualFileResult result
 
         Logger.ExecutingFileResult(result, result.FileName);
 
-        var lastModified = result.LastModified ?? fileInfo.LastModified;
+        var lastModified = fileInfo.LastModified;
+        var length = fileInfo.Length;
+
+        // If the pyshical path is available we should check if it is a symlink and
+        // get the file information from the final link target instead
+        if (!string.IsNullOrEmpty(fileInfo.PhysicalPath))
```

Comment:
Based on @Tratcher's comment, could you file an issue in the runtime repo to have [`PhysicalFileInfo`](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileInfo.cs) do the right thing? It would be the better layer to solve this in.

Reply:
Make sense. One thing to notice is the `PhysicalFileResult `and `PhysicalFileResultExecutor` do not use the PhysicalFileInfo, since they are not going thru the `FileProvider`. What do you prefer, keep my changes or they should go thru the `PhysicalFileProvider` and leverage the fix in the `PhysicalFileInfo` that we will ask?

Reply:
Keep your changes for PhysicalFileResult. Undo the changes from VirtualFileResult since they'll be resolved via changes in the runtime.

Reply:
Done. https://github.com/dotnet/runtime/issues/63424